### PR TITLE
Remove dead code from virtool.virtool

### PR DESCRIPTION
**Pull request by eroberts9789**
* Use static code analysis tools `Vulture` and `Pyflakes` to detect unused code
* Remove functions only used in test modules and not in `virtool.virtool` project code

### CODE REVIEW

File: virtool/config.py [hunk 1]
```diff
@@ -155,57 +107,6 @@ def cli(ctx, data_path, db_connection_string, db_name, dev, force_version, no_se
...
-        "no_check_db": no_check_db,
-        "no_check_files": no_check_files,
-        "no_client": no_client,
-        "no_fetching": no_fetching
-    }
-
-    logger.info("Starting in server mode")
-    asyncio.get_event_loop().run_until_complete(virtool.app.run_app(config))
-
-
```
Comment (igboyes) [inline virtool/config.py:208]: I don't think this should be removed.

File: virtool/dev/fake.py [hunk 2]
```diff
@@ -177,29 +175,6 @@ async def create_fake_analysis(app: App):
...
-        user_id=USER_ID,
-        rights=rights,
-        job_id="sample_integration_test_job",
-    )
-
-
-async def create_fake_jobs(app: App):
-    await create_sample_integration_test_job(app)
-
-
```
Comment (igboyes) [inline virtool/dev/fake.py:202]: Can this be removed @BlakeASmith?

File: virtool/logs.py [hunk 3]
```diff
@@ -41,32 +41,6 @@ def configure_base_logger(dev: bool, verbose: bool) -> Logger:
...
-        "server.log", maxBytes=1000000, backupCount=3
-    )
-
-    handler.setFormatter(logging.Formatter(get_log_format(verbose), style="{"))
-
-    logger.addHandler(handler)
-
-    return logger
-
-
```
Comment (igboyes) [inline virtool/logs.py:69]: This is called in `config.py`.

File: virtool/app.py [hunk 4]
```diff
@@ -85,21 +83,3 @@ def create_app(config):
...
-        config["port"]
-    )
-
-    _, pending = await asyncio.wait(
-        [wait_for_restart(runner, app["events"]), wait_for_shutdown(runner, app["events"])],
-        return_when=asyncio.FIRST_COMPLETED
-    )
-
-    for task in pending:
-        task.cancel()
```
Comment (igboyes) [inline virtool/app.py:105]: Called in `config.py`.

**Review by igboyes — CHANGES_REQUESTED** [review]
Some things that I believe shouldn't be removed.

I have requested input from @BlakeASmith on an bit of integration test code.

File: virtool/dev/fake.py (hunk 2 above)
Comment (BlakeASmith) [reply virtool/dev/fake.py:202]: Yes. The test job is now created in `virtool.fake.test_cases.sample_integration`

**Review by BlakeASmith — APPROVED** [review]

**Review by igboyes — APPROVED** [review]

**Pull request merged**